src/backend/distributed/commands/alter_table.c

```diff
@@ -854,8 +874,12 @@ ConvertTable(TableConversionState *con)
 				.suppressNoticeMessages = con->suppressNoticeMessages
 			};
 			TableConversionReturn *colocatedReturn = con->function(&cascadeParam);
+
+			/* persist foreign key commands to older context */
+			MemoryContextSwitchTo(oldContext);
```

Comment:
This is a no-op because the same is already done on line 845. Should line 845 be
```diff
-MemoryContextSwitchTo(oldContext);
-MemoryContextSwitchTo(convertTableContext);
```

(line 901 is also a possible no-op because of the same reason)